site/content/docs/how_the_optimizing_compiler_works/backend.md

```diff
@@ -0,0 +1,504 @@
+    RBP ----> +-----------------+                +-----------------+
+              |      `...`      |                |      `...`      |
+              |      ret Y      |                |      ret Y      |
+              |      `...`      |                |      `...`      |
+              |      ret 0      |                |      ret 0      |
+              |      arg X      |                |      arg X      |
+              |      `...`      |     ====>      |      `...`      |
+              |      arg 1      |                |      arg 1      |
+              |      arg 0      |                |      arg 0      |
+              |   Return Addr   |                |   Return Addr   |
+    RSP ----> +-----------------+                |    Caller_RBP   |
+                 (low address)                   +-----------------+ <----- RSP, RBP
+```
+
+While, on `arm64`, there is only a stack pointer `SP`:
+
+
+```goat {width="100%" height="300"}
+            (high address)                    (high address)
+  SP ---> +-----------------+               +------------------+ <----+
+          |      `...`      |               |      `...`       |      |
+          |      ret Y      |               |      ret Y       |      |
+          |      `...`      |               |      `...`       |      |
+          |      ret 0      |               |      ret 0       |      |
+          |      arg X      |               |      arg X       |      |  size_of_arg_ret.
+          |      `...`      |     ====>     |      `...`       |      |
+          |      arg 1      |               |      arg 1       |      |
+          |      arg 0      |               |      arg 0       | <----+
+          +-----------------+               |  size_of_arg_ret |
+                                            |  return address  |
+                                            +------------------+ <---- SP
+             (low address)                     (low address)
+```
+
+However, the prologue and epilogue might also be responsible for saving and
+restoring the state of registers that might be overwritten by the function
+("clobbered"); and, if spilling occurs, prologue and epilogue are also
+responsible for reserving and releasing the space for the spilled values.
+
+For clarity, we make a distinction between the space reserved for the clobbered
+registers and the space reserved for the spilled values:
+
+- Spill slots are used to temporarily store the values that needs spilling as
+  determined by the register allocator. This section must have a fix height,
+but its contents will change over time, as registers are being spilled and
+reloaded.
+- Clobbered registers are, similarly, determined by the register allocator, but
+  they are stashed in the prologue and then restored in the epilogue.
+
+The procedure happens at the end of the register allocation phase because at
+this point we have collected enough information to know how much space we need
+to reserve.
```

Comment:
```suggestion
The procedure happens after the register allocation phase because at
this point we have collected enough information to know how much space we need
to reserve, and which registers are clobbered.
```